src/js/form.js

```diff
@@ -312,27 +312,35 @@ Form.prototype.init = function () {
         this.pages.init();
 
         // after radio button data-name setting (now done in XLST)
-        // Set temporary event handler to ensure calculations in newly added repeats are run for the first time
-        const tempHandlerAddRepeat = (event) => this.calc.update(event.detail);
-        const tempHandlerRemoveRepeat = () => (this.all = {});
+        let repeatsAdded = 0;
+
+        const tempHandlerAddRepeat = () => {
+            repeatsAdded += 1;
+        };
+
         this.view.html.addEventListener(
             events.AddRepeat().type,
             tempHandlerAddRepeat
         );
-        this.view.html.addEventListener(
-            events.RemoveRepeat().type,
-            tempHandlerRemoveRepeat
-        );
+
         this.repeatsInitialized = true;
         this.repeats.init();
+
+        // Recalculate after repeats are initialized. Previously this was performed in
+        // `tempHandlerAddRepeat`, but recalculating them all at once is significantly faster.
+        if (repeatsAdded > 0) {
+            this.calc.update({
+                allRepeats: true,
+                cloned: true,
+            });
+
+            this.all = {};
```

Comment:
I played around with this to see if the events.RemoveRepeat() could fire without adding repeats first, and (like you) I believe that is not possible. So placing `this.all = {};` here and removing the remove handler seems 👍.